src/runtime_src/core/edge/drm/zocl/zocl_bo.c

```diff
@@ -808,6 +808,11 @@ int zocl_get_hbo_ioctl(struct drm_device *dev, void *data,
 		DRM_ERROR("Buffer at out side of reserved memory region\n");
 		return -ENOMEM;
 	}
+	if ((args->size & ZOCL_PAGE_SIZE_M1) != 0) {
+		//DRM requirement
+		DRM_ERROR("Buffer size must be multiple of 4096\n");
+		return -ENOMEM;
+	}
```

Comment:
1) I would suggest using system macro instead of define our own. PAGE_ALIGEND might be your best candidate
2) We probably want to check the start address and size are both PAGE_ALIGEND and also the size should not be zero, if it is not checked somewhere else.
3) The return error code should be -EINVAL not -ENOMEM

Reply:
done